pkg/scanner/scanner.go

```diff
@@ -81,13 +82,37 @@ func (p *scanParser) checkPortsRemoved(host string) {
 	}
 }
 
-type nmapWrapper struct{}
+type nmapWrapper struct {
+	interfaceName string
+}
 
 func (n *nmapWrapper) Run(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
+	if n.interfaceName != "" {
+		return n.runWithDevice(ipAddresses, ctx)
+	}
+	return n.runWithoutDevice(ipAddresses, ctx)
+}
+
+func (n *nmapWrapper) runWithDevice(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
```

Comment:
Tiny nit, but you can DRY this up a bit.  `nmap.NewScanner()` takes a slice of `func(*Scanner)` references as input.  You could build the slice in each of the `runXXXDevice` methods and then call another function that calls the `nmap.NewScanner()` function, handles the error and runs the scanner.  Something like: 

```golang
func (n *nmapWrapper) runWithDevice(ipAddresses []string, ctx context.Context) (result *nmap.Run, warnings []string, err error) {
  return runScanner(
    ipAddress,
    ctx,
    nmap.WithTargets(ipAddresses...),
    nmap.WithContext(ctx),
    nmap.WithSkipHostDiscovery(),
    nmap.WithInterface(n.interfaceName),
    nmap.WithUnprivileged(),
  )
}

func runScanner(ipAddresses []string, ctx context.Context, options ...func(*Scanner)) (result *nmap.Run, warnings []string, err error)
  scanner, err := nmap.NewScanner(options...)
  if err != nil {
    return nil, nil, fmt.Errorf("unable to create scanner scanner: %v", err)
  }
  return scanner.Run()
}
```

Reply:
Good call. I refactored it.